torchrl/envs/model_based.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
Seems like some of the changes in this PR belong in #333? It would be cleaner for code review and log history to have this PR only include the dreamer specific implementations of the abstractions provided in #333.